File: repositories/prescription_repository.go
package repositories

import (
	"context"

	"doctor-service/entities"

	"github.com/google/uuid"
	"gorm.io/gorm"
)

type PrescriptionRepository interface {
	Create(ctx context.Context, p *entities.Prescription) error
	FindByDoctorID(ctx context.Context, doctorID uuid.UUID) ([]entities.Prescription, error)
	FindByID(ctx context.Context, id uuid.UUID) (*entities.Prescription, error)
}

type GormPrescriptionRepository struct {
	db *gorm.DB
}

func NewPrescriptionRepository(db *gorm.DB) PrescriptionRepository {
	return &GormPrescriptionRepository{db: db}
}

func (r *GormPrescriptionRepository) Create(ctx context.Context, p *entities.Prescription) error {
	return r.db.WithContext(ctx).Create(p).Error
}

func (r *GormPrescriptionRepository) FindByDoctorID(ctx context.Context, doctorID uuid.UUID) ([]entities.Prescription, error) {
	var list []entities.Prescription
	err := r.db.WithContext(ctx).Where("doctor_id = ?", doctorID).Order("issued_at DESC").Find(&list).Error
	return list, err
}

func (r *GormPrescriptionRepository) FindByID(ctx context.Context, id uuid.UUID) (*entities.Prescription, error) {
	var p entities.Prescription
	err := r.db.WithContext(ctx).Where("prescription_id = ?", id).First(&p).Error
	return &p, err
}

File: python_backend/repositories/prescription_repository.py
from typing import List, Optional
from uuid import UUID
from sqlalchemy.orm import Session
from entities.prescription import Prescription

class PrescriptionRepository:
    def __init__(self, db: Session):
        self.db = db

    def Create(self, p: Prescription) -> None:
        self.db.add(p)
        self.db.commit()
        self.db.refresh(p)

    def FindByDoctorID(self, doctor_id: UUID) -> List[Prescription]:
        return self.db.query(Prescription).filter(Prescription.doctor_id == doctor_id).order_by(Prescription.issued_at.desc()).all()

    def FindByID(self, id: UUID) -> Optional[Prescription]:
        return self.db.query(Prescription).filter(Prescription.prescription_id == id).first()
